include/shell/shell_uart.h

```diff
@@ -81,6 +81,34 @@ struct shell_uart {
  */
 const struct shell *shell_backend_uart_get_ptr(void);
 
+/**
+ * @typedef shell_backend_uart_fork_rx_callback_t
+ * @brief Define the RX callback when forking the shell UART via
+ * shell_backend_uart_fork_rx()
+ *
+ * @param dev		UART device structure.
+ * @param user_data	Pointer to data specified by user.
+ * @param data		Pointer to the start of the data coming from RX (this
+ *			address range will be recycled, so data should be
+ *			copied to persist).
+ * @param length	Number of bytes available starting at {@code data}.
+ */
+typedef void (*shell_backend_uart_fork_rx_callback_t)(const struct device *dev,
```

Comment:
Can you please find a shorter name for this?

Reply:
Done